x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyReindexPipeline.java

```diff
@@ -68,7 +70,19 @@ static boolean exists(ClusterState clusterState) {
      */
     public static void create(Client client, ActionListener<AcknowledgedResponse> listener) {
         final BytesReference pipeline = BytesReference.bytes(currentEnrichPipelineDefinition(XContentType.JSON));
-        client.execute(PutPipelineTransportAction.TYPE, new PutPipelineRequest(pipelineName(), pipeline, XContentType.JSON), listener);
+        client.execute(
+            PutPipelineTransportAction.TYPE,
+            new PutPipelineRequest(
+                // wait for as long as it takes to create the pipeline
+                MasterNodeRequest.INFINITE_MASTER_NODE_TIMEOUT,
+                // TODO should this be longer?
+                AcknowledgedRequest.DEFAULT_ACK_TIMEOUT,
```

Comment:
This is the only questionable bit really, should these timeouts both be infinite? Or zero for the ack timeout?

Reply:
is it bad to leave it as the default of 30 seconds? Do we need to decide between 0 or infinite?

Reply:
No, it can be whatever we want, if we decide they should remain at 30s then that's fine too. IMO 30s is fairly suspicious tho, it means that if this happens to be running in a cluster that's a little slow then this will fail with a timeout, and it's not a timeout that the user can influence. But I don't have any strong opinions here.

Reply:
I'm not sure if we should change it then. I suppose I'm afraid that changing it to be infinite without a way to cancel the enrichment execution would be bad, and changing it to 0 so that there's zero tolerance for a cluster to have any delay would also be bad. I don't have a strong opinion though.

Reply:
Ok I've set it back to 30s, introducing a new enrich-specific constant to make it clearer that this is a positive decision.